src/app/WeaponPicker.tsx

```diff
@@ -0,0 +1,65 @@
+import { Autocomplete, Box, TextField, Typography } from "@mui/material";
+import { memo, useCallback, useMemo } from "react";
+import type { Weapon } from "../calculator/weapon";
+import { weaponTypeLabels } from "./uiUtils";
+
+export type WeaponOption = {
+  label: string; // weaponName
+  value: string; // weaponName
+  type: string; // weaponType
+};
+
+interface Props {
+  selectedWeapons: WeaponOption[];
+  onSelectedWeaponsChanged(weapons: WeaponOption[]): void;
+  weaponOptions: WeaponOption[];
+}
+
+const sortByTypeThenName = (a: Weapon, b: Weapon): number => {
+  if (a.weaponType < b.weaponType) return -1;
+  if (a.weaponType > b.weaponType) return 1;
+  // weapon types are the same, so compare the name
+  if (a.weaponName < b.weaponName) return -1;
+  if (a.weaponName > b.weaponName) return 1;
+  // both primary and secondary values are equal
+  return 0;
+};
+const makeOption = (weapon: Weapon): WeaponOption => ({
+  label: weapon.weaponName,
+  value: weapon.weaponName,
+  type: weaponTypeLabels.get(weapon.weaponType) || "",
+});
+
+export const makeWeaponOptionsFromWeapon = (weapons: Weapon[]): WeaponOption[] =>
+  weapons.sort(sortByTypeThenName).map(makeOption);
```

Comment:
I think we should make this filter out DLC weapons if `includeDLC` is false, for people who don't own the DLC or are avoiding spoilers

Reply:
Should be able to pass `includeDLC` from App, and filter by `weapon.dlc`

Reply:
That's a great point! I'll include that.